gtfs-realtime/spec/en/reference.md

```diff
@@ -1,14 +1,29 @@
 A GTFS Realtime feed lets transit agencies provide consumers with realtime information about disruptions to their service (stations closed, lines not operating, important delays, etc.) location of their vehicles, and expected arrival times.
 
-Version 1.0 of the feed specification is discussed and documented on this site.
+Version 2.0 of the feed specification is discussed and documented on this site.
 
 ### Term Definitions
 
-*   **required**: Exactly one
-*   **repeated**: Zero or more
```

Comment:
Sounds good on adding the new column.

So here's what I'm working on now to update this proposal, in summary:

* `Required` column will address semantic requirements
* `Cardinality` column will include information about how many elements should be provided for a particular field

The goal of this proposal is to represent the semantic requirements of fields, based on transit use cases and domain logic.  The main challenge here is that there isn't a direct mapping between semantic requirements in `Required` and protobuf cardinality.  As a result I think it will be confusing to show semantic requirements in `Required` and protobuf cardinality in `Cardinality`.

I'd prefer to keep the spec documentation independent of the implementation - in other words, the docs should be purely semantic, with the `.proto` file being the reference for the protobuf cardinality.

With this in mind, I propose that we define semantic cardinality something like the following:

> *Cardinality* represents the number of elements that may be provided for a particular field:
>
>* **One** - A single one element may be provided for this field.  This maps to the [Protocol Buffer *required* and *optional* cardinalities](https://developers.google.com/protocol-buffers/docs/proto#simple).
>* **Many** - Many elements (0, 1, or more) may be provided for this field.  This maps to the [Protocol Buffer *repeated* cardinality](https://developers.google.com/protocol-buffers/docs/proto#simple).
>
>Always reference the *Required* and *Description* fields to see when a field is required, conditionally required, or optional.  Please reference [`gtfs-realtime.proto`](https://github.com/google/transit/blob/master/gtfs-realtime/proto/gtfs-realtime.proto) for Protocol Buffer cardinality.

Using these definitions, `FeedMessage` and `TripUpdate` would look like the following:

#### FeedMessage

| _**Field Name**_ | _**Type**_ | _**Required**_ | _**Cardinality**_ | _**Description**_ |
|------------------|------------|----------------|-------------------|-------------------|
|**header** | [FeedHeader](#message-feedheader) | Required | One | Metadata about this feed and feed message. |
|**entity** | [FeedEntity](#message-feedentity) | Required | Many | Contents of the feed. |

#### TripUpdate

| _**Field Name**_ | _**Type**_ | _**Required**_ | _**Cardinality**_ | _**Description**_ |
|------------------|------------|----------------|-------------------|-------------------|
| **trip** | [TripDescriptor](#message-tripdescriptor) | Required | One | XXXX |
| **vehicle** | [VehicleDescriptor](#message-vehicledescriptor) | Optional | One | XXXX |
| **stop_time_update** | [StopTimeUpdate](#message-stoptimeupdate) | Conditionally required | Many | XXXX |
| **timestamp** | [uint64](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Optional | One | XXXX |
| **delay** | [int32](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Optional | One | XXXX |